### PR TITLE
Prevent IMAP passwords from going to the logging system

### PR DESCRIPTION
**Pull request by NicolasLM**
Even if the final user did not enable DEBUG logging for its
application, some logs records can end up in tools like Sentry
via breadcrumbs.

This commit makes sure that passwords are never sent to the
logging system by verifying and redacting the log message before
passing it to the logger.

In the future, once we get rid of imaplib, the code replacing it
could avoid logging raw data containing sensitive information in
the first place.

### CODE REVIEW

**Comment by mlorant** [comment]
Awesome, I did something equivalent because of Sentry too. However, could you also support `oauth2_login` and strip the user and access_token in that case? Thanks!

**Comment by NicolasLM** [comment]
I added the same thing for the `AUTHENTICATE` command.

File: imapclient/test/test_imapclient.py [hunk 1]
```diff
@@ -365,6 +366,26 @@ def test_IMAP_is_patched(self):
...
+            "> b'ICHH1 LOGIN **REDACTED**",
+            (),
+            extra={}
+        )
+        py2_called = logger_mock.info.called_once_with(
+            "> b'ICHH1 LOGIN **REDACTED**",
+            (),
+            extra={}
+        )
+        assert py2_called or py3_called, "Log message not redacted properly"
```
Comment (mjs) [inline imapclient/test/test_imapclient.py]: Why not use `six.PY3` to determine the Python version and choose the assert?

Comment (NicolasLM) [inline imapclient/test/test_imapclient.py]: Done

File: imapclient/test/test_imapclient.py [hunk 2]
```diff
@@ -365,6 +366,27 @@ def test_IMAP_is_patched(self):
...
+        if six.PY3:
+            logger_mock._log.assert_called_once_with(
+                logging.INFO,
+                "> b'ICHH1 LOGIN **REDACTED**",
+                (),
+                extra={}
+            )
+        else:
+            logger_mock.info.assert_called_once_with(
+                "> b'ICHH1 LOGIN **REDACTED**",
```
Comment (mlorant) [inline imapclient/test/test_imapclient.py]: no `b'` prefix in py2, but I think the CI error would have help you too

Comment (NicolasLM) [inline imapclient/test/test_imapclient.py]: Fixed, thanks!

**Comment by mjs** [comment]
I'm happy to merge this but there's a conflict.

**Comment by NicolasLM** [comment]
Rebased

**Comment by mjs** [comment]
Thanks again.

**Pull request merged**